### DISCUSSION TITLE
Two argocd in two namespace can view each other namespace deployments?

### DISCUSSION THREAD

**Discussion by BinaryWorld2099** [argoproj/argo-cd#10819] [Ideas]
I have two git repositories, one for testing, one for development in single k8s cluster for QA and dev teams

cmd: 
helm install test-argocd -n test-argocd --create-namespace
helm install dev-argocd -n dev-argocd --create-namespace

namespace test, install an argocd

namespace dev, install an argocd

But test namespace argocd can view deployment in dev namespace argocd, vice versa

seems argocd can view all deployment in all namespaces.

Is it possible to isolate the two argocd in their namespace only?

Is it possible to install two argocd complete separately include deployment, service account, role binding in different namesapces?

Another issue I faced was that, after successfully install test-argocd and dev-argocd
with kubectl get pod -n test-argocd, it seems deploy two sets of argocd like below in test-argocd and dev-argocd

in test-argocd namespace: 
test-argocd-application-controller-0    
test-argocd-applicationset-controller-3gh6e4r-dwt2w
test-argocd-dex-server-gese45-x8dz8
test-argocd-notifications-controller-gerg6t352-n9lxn
test-argocd-redis-fsef7y67-te4gr
test-argocd-repo-server-654rgegr-jygjy
test-argocd-server-gers45-54165
argocd-application-controller-0    
argocd-applicationset-controller-6676c4d655-dwt2w
argocd-dex-server-5f9bd56758-x8dz8
argocd-notifications-controller-5bc766fc6d-n9lxn
argocd-redis-6fd7cbd95d-tltqz
argocd-repo-server-5bb545d667-vcjgz
argocd-server-6b88fbcbb8-j4kf7  

It is because of the selfheal healing?

**Comment by jgwest** [comment]
> Is it possible to install two argocd complete separately include deployment, service account, role binding in different namesapces?

Yes, see the `namespace-install.yaml` section [here](https://github.com/argoproj/argo-cd/tree/master/manifests). 

e.g:
```bash
kubectl create namespace argocd-one
kubectl apply -n argocd-one -f namespace-install.yaml
```

With this method, you may need to setup `Role` and `RoleBindings` for the serviceaccount Argo CD uses to deploy (`argocd cluster add <CONTEXT> --in-cluster --namespace <YOUR NAMESPACE>)` may do this for you, but I'm not sure)

**Reply by BinaryWorld2099** [reply]
I am using Helm to install argocd,
checked the arogocd helm chart template with  Cluster Role.
Is it because the Cluster Role making two namespace argocd can look at each other deployment?
Does argocd helm support namespace installation?
![image](https://user-images.githubusercontent.com/106383882/194573863-8d5a4edc-a1a5-45cb-b331-d2dfcca27d3b.png)

**Reply by jgwest** [reply]
The argo-helm chart is community maintained, I'm personally not familiar with how it is constructed. You wish to check if it supports namespace-scoped installation.
